functions/src/utils/permissions.ts

```diff
@@ -62,4 +138,40 @@ export const checkIsWorldOwner = async (worldId: string, uid: string) => {
   }
 };
 
-exports.checkIsAdmin = checkIsAdmin;
+export const throwErrorIfNotWorldOwner = async ({
+  worldId,
+  uid,
+}: {
+  worldId: string;
+  uid: string;
+}) => {
+  const isWorldOwner = await checkIfWorldOwner({ worldId, uid });
+
+  if (!isWorldOwner) {
+    throw new HttpsError(
+      "permission-denied",
+      `User is not an owner of ${worldId}`
+    );
+  }
+};
+
+export const throwErrorIfNeitherWorldNorSpaceOwner = async ({
```

Comment:
This might be better named `throwIfCannotEditSpace`

Reply:
I don't think it's a better name though. This function's name exactly describes what it does. `throwIfCannotEditSpace` on the other hand is a specific usecase where it is used. We would probably want to use this function in the delete messages cloud function(if it was implemented via functions). Then, it wouldn't make any sense to call `throwIfCannotEditSpace` as we would not edit the space